packages/suite/src/hooks/wallet/useCoinmarketSellOffers.ts

```diff
@@ -0,0 +1,301 @@
+                getQuotes();
+            }
+        }
+    });
+
+    const doSellTrade = async (quote: SellFiatTrade) => {
+        if (!quotesRequest) return;
+        const returnUrl = await createQuoteLink(quotesRequest, account);
+        const response = await invityAPI.doSellTrade({ trade: quote, returnUrl });
+        if (response) {
+            if (response.trade.error) {
+                console.log(`[doSellTrade] ${response.trade.error}`);
+                addNotification({
+                    type: 'error',
+                    error: response.trade.error,
+                });
+                return undefined;
+            }
+            if (
+                response.trade.status === 'LOGIN_REQUEST' ||
+                response.trade.status === 'SITE_ACTION_REQUEST'
+            ) {
+                submitRequestForm(response.tradeForm?.form);
+                return undefined;
+            }
+            return response.trade;
+        }
+        const errorMessage = 'No response from the server';
+        console.log(`[doSellTrade] ${errorMessage}`);
+        addNotification({
+            type: 'error',
+            error: errorMessage,
+        });
+    };
+
+    const selectQuote = async (quote: SellFiatTrade) => {
+        const provider =
+            sellInfo?.providerInfos && quote.exchange
+                ? sellInfo.providerInfos[quote.exchange]
+                : null;
+        if (quotesRequest) {
+            const result = await openCoinmarketSellConfirmModal(provider?.companyName);
+            if (result) {
+                // empty quoteId means the partner requests login first, requestTrade to get login screen
+                if (!quote.quoteId) {
+                    doSellTrade(quote);
+                } else {
+                    setSelectedQuote(quote);
+                    timer.stop();
+                }
+            }
+        }
+    };
+
+    const confirmTrade = async (bankAccount: BankAccount) => {
+        if (!selectedQuote) return;
+        setCallInProgress(true);
+        const quote = { ...selectedQuote, bankAccount };
+        const response = await doSellTrade(quote);
+        if (response) {
+            setSelectedQuote(response);
+            setSellStep('SEND_TRANSACTION');
+        }
+        setCallInProgress(false);
+    };
+
+    const addBankAccount = async () => {
+        if (!selectedQuote) return;
+        setCallInProgress(true);
+        await doSellTrade(selectedQuote);
+        setCallInProgress(false);
+    };
+
+    const sendTransaction = async () => {
+        if (
+            selectedQuote &&
+            selectedQuote.orderId &&
+            selectedQuote.destinationAddress &&
+            selectedQuote.cryptoStringAmount &&
+            composed
+        ) {
+            // prepare the fee levels, set custom values from composed
+            // WORKAROUND: sendFormEthereumActions and sendFormRippleActions use form outputs instead of composed transaction data
```

Comment:
discuss

Reply:
Refactored to useCoinmarketRecomposeAndSign hook in sell and exchange crypto.